presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationMetadata.java

```diff
@@ -147,7 +150,8 @@ private static void verifyInputFunctionSignature(MethodHandle method, List<Param
     private static void verifyCombineFunction(MethodHandle method, Class<?> stateInterface)
     {
         Class<?>[] parameterTypes = method.type().parameterArray();
-        checkArgument(parameterTypes.length == 2 && parameterTypes[0] == stateInterface && parameterTypes[1] == stateInterface, "Combine function must have the signature (%s, %s)", stateInterface.getSimpleName(), stateInterface.getSimpleName());
+        checkArgument(parameterTypes.length >= 2, "Combine function must take at least 2 arguments.");
+        checkArgument(Arrays.stream(parameterTypes).filter(type -> type.equals(stateInterface)).count() == 2, "Combine function must take exactly two arguments of type %s annotated as @AggregationState", stateInterface.getSimpleName());
```

Comment:
I understand that this is after the method handle is bound. In this case, why would you have more than 2 arguments?